python/paddle/fluid/dygraph/dygraph_to_static/loop_transformer.py

```diff
@@ -217,21 +253,30 @@ def while_loop_body(self.x, y) # `self.x` is illegal.
         if attr_full_name.startswith("self."):
             return
         self.current_seen_vars.add(node)
+
+        write_context = {
+            type(gast.Store()), type(gast.AugStore()), type(gast.Del())
+        }
         for loop_node in self.current_loop:
             self.in_loop_vars[loop_node].add(node)
+
         # sub-nodes are visited during get_attribute_full_name and we shouldn't
         # visit again
 
     def visit_For(self, node):
         self.current_loop.append(node)
+        self.in_condition = True
         self.visit(node.target)
```

Comment:
Doesn't `node.iter` also need to visit ? 

Because in the following code, `iter_num` may be not in `loop_vars`, but it should be.
```Python
def fn(x):
  iter_num = 3
  for i in range(iter_num):
     x = x+1
  return x
```

Reply:
You are right, I will add it.